src/main/java/org/corpus_tools/peppermodules/rstModules/RST2SaltMapper.java

```diff
@@ -433,9 +478,112 @@ private void mapSignal(Signal signal, SLayer layer) {
 				tokRel.setTarget(sTokens.get(tokenId - 1));
 				tokRel.setType("signal_token");
 				tokRel.setSource(signalNode);
-				layer.addRelation(tokRel);
 				this.getDocument().getDocumentGraph().addRelation(tokRel);
 			}
 		}
+
+		//////////////////////////////////////////////////////////////////////////////////
+		// Signal to other node relation creation
+		//////////////////////////////////////////////////////////////////////////////////
+		if (secondaryEdge == null) {
+			// make the signal node dominate the RST node that it is associated with. When
+			// RST is represented graphically, this is the node that the arrow points out of,
+			// but in Salt, it is the node that is the child of a dominance relation
+			SDominanceRelation signal2rstNode = SaltFactory.createSDominanceRelation();
+			String associatedSignalNodeId = signal.getSource().getId();
+			signal2rstNode.setSource(signalNode);
+			signal2rstNode.setTarget(sSource);
+
+			// If the discourse unit associated with this signal has a parent (i.e., it is
+			// the satellite of a nucleus), then we want to add some additional annotations.
+			// Specifically, we want to add the name of the relation to the signal relation,
+			// and we want to annotate teh signal node with the name of the relation.
+			List<Relation> incomingRelations = this.getCurrentRSTDocument()
+					.getIncomingRelations(associatedSignalNodeId);
+			if (incomingRelations != null && incomingRelations.size() > 0) {

```

Comment:
This is a common pattern in the original code, but since getting the size can be costly, just using `isEmpty()` is more efficient.

```suggestion
			if (incomingRelations != null && incomingRelations.isEmpty()) {

```